src/app/user/application/usecases/delete-user-account.use-case.ts

```diff
@@ -0,0 +1,24 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
+import { IUserRepository } from '@app/user/domain/ports/iuser.repository';
+
+@Injectable()
+export class DeleteUserAccountUseCase {
+  constructor(@Inject('UsersRepository') private readonly userRepository: IUserRepository) {}
+
+  async execute(userId: string): Promise<void> {
```

Comment:
```suggestion
  async execute(user: User): Promise<void> {
```
استخدم CurrentUser من application/deocrators

Reply:
Done